sdks/python/apache_beam/ml/rag/ingestion/base.py

```diff
@@ -0,0 +1,112 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+from abc import abstractmethod
+
+import apache_beam as beam
+from apache_beam.ml.rag.types import Chunk
+
+
+class VectorDatabaseWriteConfig(ABC):
+  """Abstract base class for vector database configurations in RAG pipelines.
+  
+  VectorDatabaseWriteConfig defines the interface for configuring vector
+  database writes in RAG pipelines. Implementations should provide
+  database-specific configuration and create appropriate write transforms.
+
+  The configuration flow:
+  1. Subclass provides database-specific configuration (table names, etc)
+  2. create_write_transform() creates appropriate PTransform for writing
+  3. Transform handles converting Chunks to database-specific format
+
+  Example implementation:
+    >>> class BigQueryVectorWriterConfig(VectorDatabaseWriteConfig):
+    ...     def __init__(self, table: str):
+    ...         self.embedding_column = embedding_column
+    ...         
+    ...     def create_write_transform(self):
+    ...         return beam.io.WriteToBigQuery(
+    ...             table=self.table
+    ...         )
+  """
+  @abstractmethod
+  def create_write_transform(self) -> beam.PTransform:
+    """Creates a PTransform that writes embeddings to the vector database.
+    
+    Returns:
+        A PTransform that accepts PCollection[Chunk] and writes the chunks'
+        embeddings and metadata to the configured vector database.
+        The transform should handle:
+        - Converting Chunk format to database schema
+        - Setting up database connection/client
+        - Writing with appropriate batching/error handling
+    """
+    pass
```

Comment:
Nit: It is probably better to throw here like https://github.com/apache/beam/blob/a51a0e1688978cd5c90e1de4fc7616ff068eb7b4/sdks/python/apache_beam/ml/inference/base.py#L165 - it will provide more intelligible errors when a user messes something up (e.g. a typo in the override)